go/pkg/client/client.go

```diff
@@ -401,13 +455,16 @@ func getImpersonatedRPCCreds(ctx context.Context, actAs string, cred credentials
 	}
 }
 
-func getRPCCreds(ctx context.Context, credFile string, useApplicationDefault bool, useComputeEngine bool) (credentials.PerRPCCredentials, error) {
+func getRPCCreds(ctx context.Context, credFile string, useApplicationDefault bool, useComputeEngine bool, authUsed *AuthType) (credentials.PerRPCCredentials, error) {
```

Comment:
Why not return the AuthType?

Reply:
I initially thought about it, but it involved a bit of change to predeclare variables and I figured just passing a pointer would make it easier. But now that I think about it, it seems cleaner and standardized to keep return values as return values instead of as pointers passed to the function. Changed!